docs/contracts/StMatic.md

```diff
@@ -406,233 +307,109 @@ Requirements:
+```solidity
+function claimTokens(uint256 _tokenId)
 ```
-
 #### Parameters
 
-| Name             | Type      | Description                                         |
-| ---------------- | --------- | --------------------------------------------------- |
-| `_insuranceFund` | `address` | Address of contract which accumulates insurance fee |
-
-### setWithdrawalCredentials()
-
-Set credentials to withdraw ETH on ETH 2.0 side after the phase 2 is launched to `_withdrawalCredentials`
+| Name              | Type     | Description                                                    |
+| ----------------- | -------- | -------------------------------------------------------------- |
+| `_tokenId` | `uint256` | Token ID of a pending withdrawal request. |
+---
 
-```sol
-function setWithdrawalCredentials(bytes32 _withdrawalCredentials)
+### distributeRewards()
+Distributes rewards claimed from validators to their operators
+```solidity
+function distributeRewards()
 ```
+---
 
-:::note
-Note that `setWithdrawalCredentials` discards all unused signing keys as the signatures are invalidated.
-:::
-
-#### Parameters
-
-| Name                     | Type      | Description                                                                                 |
-| ------------------------ | --------- | ------------------------------------------------------------------------------------------- |
-| `_withdrawalCredentials` | `bytes32` | Hash of withdrawal multisignature key as accepted by the deposit_contract.deposit functione |
-
-### withdraw()
-
-Issues withdrawal request. **Not implemented.**
-
-```sol
-function withdraw(uint256 _amount, bytes32 _pubkeyHash)
+### withdrawTotalDelegated()
+Allows a node operator withdraw the amount staked to a validator
+```solidity
+function withdrawTotalDelegated(address _validatorShare)
 ```
-
-:::note
-Will be upgraded to an actual implementation when withdrawals are enabled (Phase 1.5 or 2 of Eth2 launch, likely late 2021 or 2022). At the moment withdrawals are not possible in the beacon chain and there's no workaround
-:::
-
 #### Parameters
 
-| Name          | Type      | Description                 |
-| ------------- | --------- | --------------------------- |
-| `_amount`     | `uint256` | Amount of StETH to withdraw |
-| `_pubkeyHash` | `bytes32` | Receiving address           |
-
-### pushBeacon()
-
-Updates the number of Lido-controlled keys in the beacon validators set and their total balance.
+| Name              | Type     | Description                                                    |
+| ----------------- | -------- | -------------------------------------------------------------- |
+| `_validatorShare` | `address` | Address of the validator proxy. |
+---
 
-```sol
-function pushBeacon(uint256 _beaconValidators, uint256 _beaconBalance)
+### claimTokens2StMatic()
+Claims tokens frm the validator and sends them to the StMatic contract
```

Comment:
Claim and transfer the tokens from a spécific validator share contract to the StMATIC contract. This requires a valid Lido NFT token.